src/app/pages/BlocksPage/index.tsx

```diff
@@ -52,26 +72,51 @@ export const BlocksPage: FC = () => {
             },
           }
         },
+        keepPreviousData: tableView === TableView.Vertical,
       },
     },
   )
 
   return (
-    <PageLayout>
+    <PageLayout
+      mobileFooterAction={<LoadMoreButton pagination={pagination} isLoading={blocksQuery.isLoading} />}
+    >
       {!isMobile && <Divider variant="layout" />}
-      <SubPageCard title={t('blocks.latest')}>
-        <Blocks
-          isLoading={blocksQuery.isLoading}
-          blocks={blocksQuery.data?.data.blocks}
-          limit={PAGE_SIZE}
-          verbose={true}
-          pagination={{
-            selectedPage: pagination.selectedPage,
-            linkToPage: pagination.linkToPage,
-            totalCount: blocksQuery.data?.data.total_count,
-            rowsPerPage: NUMBER_OF_ITEMS_ON_SEPARATE_PAGE,
-          }}
-        />
+      <SubPageCard
+        title={t('blocks.latest')}
+        action={isMobile && <TableViewSpeedDial tableView={tableView} setTableView={setTableView} />}
+        noPadding={tableView === TableView.Vertical}
+      >
+        {tableView === TableView.Horizontal && (
+          <Blocks
+            isLoading={blocksQuery.isLoading}
+            blocks={blocksQuery.data?.data.blocks}
+            limit={PAGE_SIZE}
+            verbose={true}
+            pagination={{
+              selectedPage: pagination.selectedPage,
+              linkToPage: pagination.linkToPage,
+              totalCount: blocksQuery.data?.data.total_count,
+              rowsPerPage: NUMBER_OF_ITEMS_ON_SEPARATE_PAGE,
+            }}
+          />
+        )}
+        {tableView === TableView.Vertical && (
+          <BlockDetails>
```

Comment:
It bothers me that these vertical cards aren't just a different layout of a table. They are a complete second implementation of Blocks list and of Transactions list. It doubles complexity of every view that supports it.

Some limitations if it was implemented as a Table feature instead (converting columns into rows):
- Cards would have to show the same information as the table shows. No new information like transaction's `Gas limit` that is not a column
- And show it in the same order as columns are
- Pagination
  - It would still have to leak implementation details into parent queries (limit, offset)
  - `Show More` could not be part of the footer